src/sql/src/plan/query.rs

```diff
@@ -1588,15 +1570,48 @@ fn plan_view_select(
         relation_expr = relation_expr.filter(vec![expr]);
     }
 
-    // Step 3. Gather aggregates.
-    let aggregates = {
-        let mut aggregate_visitor = AggregateFuncVisitor::new(&qcx.scx);
-        aggregate_visitor.visit_select(&s);
-        for o in order_by_exprs {
-            aggregate_visitor.visit_order_by_expr(o);
+    // Step 3. Gather aggregates and table functions.
+    let (aggregates, table_funcs) = {
+        let mut visitor = AggregateTableFuncVisitor::new(&qcx.scx);
+        visitor.visit_select_mut(&mut s);
+        for o in order_by_exprs.iter_mut() {
+            visitor.visit_order_by_expr_mut(o);
         }
-        aggregate_visitor.into_result()?
+        visitor.into_result()?
     };
+    let mut table_func_names: HashMap<String, Ident> = HashMap::new();
+    if !table_funcs.is_empty() {
+        let rows_from_qcx =
+            qcx.derived_context(from_scope.clone(), qcx.relation_type(&relation_expr));
+        let (expr, mut scope, num_cols) =
+            plan_rows_from_internal(&rows_from_qcx, table_funcs.keys(), None)?;
+        for (table_func, id) in table_funcs.iter() {
+            table_func_names.insert(id.clone(), table_func.name.0.last().unwrap().clone());
+        }
+        // Munge the scope so table names match with the generated ids.
+        let mut i = 0;
```

Comment:
100% immaterial and maybe this is even intentional but you can use `.enumerate()` on the `zip()` to get this counting done for free.